node/actors/consensus/src/replica/leader_prepare.rs

```diff
@@ -1,11 +1,73 @@
 use super::StateMachine;
-use crate::{inner::ConsensusInner, replica::error::Error};
+use crate::inner::ConsensusInner;
+use anyhow::Context as _;
 use concurrency::ctx;
 use network::io::{ConsensusInputMessage, Target};
 use roles::validator;
 use std::collections::HashMap;
 use tracing::instrument;
 
+#[derive(thiserror::Error, Debug)]
+#[allow(clippy::missing_docs_in_private_items)]
+pub(crate) enum Error {
+    #[error("bad protocol version")]
+    BadProtocolVersion,
+    #[error("invalid leader (correct leader: {correct_leader:?}, received leader: {received_leader:?}])")]
+    InvalidLeader {
+        correct_leader: validator::PublicKey,
+        received_leader: validator::PublicKey,
+    },
+    #[error("message for a past view/phase (current view: {current_view:?}, current phase: {current_phase:?})")]
+    Old {
+        current_view: validator::ViewNumber,
+        current_phase: validator::Phase,
+    },
+    #[error("invalid signature: {0:#}")]
+    InvalidSignature(#[source] crypto::bls12_381::Error),
+    #[error("invalid PrepareQC: {0:#}")]
+    InvalidPrepareQC(#[source] anyhow::Error),
+    #[error("invalid high QC: {0:#}")]
+    InvalidHighQC(#[source] anyhow::Error),
+    #[error(
+        "high QC of a future view (high QC view: {high_qc_view:?}, current view: {current_view:?}"
+    )]
+    HighQCOfFutureView {
+        high_qc_view: validator::ViewNumber,
+        current_view: validator::ViewNumber,
+    },
+    #[error("new block proposal when the previous proposal was not finalized")]
+    ProposalWhenPreviousNotFinalized,
+    #[error("block proposal with invalid parent hash (correct parent hash: {correct_parent_hash:#?}, received parent hash: {received_parent_hash:#?}, block: {header:?})")]
```

Comment:
Nit: Like with other string literals, it's possible to split these using `\`-escaped newlines; it'll consume the whitespace at the start of the next line.

Reply:
I don't really care either way, but it's kind of a burden to do it manually. Is there an autoformat option that we could use for that?

Reply:
> but it's kind of a burden to do it manually

Not a burden for me, so I'll fix this separately.